src/imitation/algorithms/base.py

```diff
@@ -166,6 +167,16 @@ def set_demonstrations(self, demonstrations: AnyTransitions) -> None:
                 `TransitionKind` instance, or a Sequence of Trajectory objects.
         """
 
+    # TODO(adam): this will cause minor headache for MCE IRL and density baseline
+    # which do not train a policy by default. Could return None for these?
+    # Or just have them train when calling this. Not so bad for MCE IRL
+    # (can compute it analytically in reasonable time frame), more awkward
+    # for density baseline.
```

Comment:
I wanted `.policy()` so that we can have a consistent interface, with the same way to evaluate the resulting policy across all algorithms. However, the interface is different enough in other ways I'm not sure how advantageous this ends up being. Implementing `policy()` turned out OK for MCE and density, so I'm inclined to leave this as-is, but I think it'd be reasonable to switch to `NotImplementedError` in the future if we have methods where supporting this is awkward.